linkerd/announcer/curatorsd/src/main/scala/com/medallia/l5d/curatorsd/announcer/CuratorSDAnnouncer.scala

```diff
@@ -3,85 +3,72 @@ package com.medallia.l5d.curatorsd.announcer
 import java.net.InetSocketAddress
 
 import com.medallia.l5d.curatorsd.common.{CuratorSDCommon, ServiceDiscoveryInfo}
-import com.medallia.servicediscovery.ServiceInstanceInfo
+import com.medallia.servicediscovery.ServiceDiscoveryRegistrar.RegistrationFormat
+import com.medallia.servicediscovery.ServiceId
 import com.twitter.finagle.{Announcement, Path}
 import com.twitter.logging.Logger
 import com.twitter.util.{Future, Time}
 import io.buoyant.linkerd.FutureAnnouncer
-import org.apache.curator.x.discovery._
 
 /**
  * Announcer that uses the curator service discovery format.
  * <p>
  * Format used for the service name Path:
- * /#/com.medallia.curatorsd/protocol/[tenant]/service_name
+ * /#/com.medallia.curatorsd/protocol/[environment]/[tenant]/service_name
  *
- * tenant is optional, "_" represents multitenant services
+ * <ul>
+ *   <li>tenant is optional, "_" represents multitenant services
+ *   <li>environment is optional, "_" represents global services
+ * </ul>
  */
-class CuratorSDAnnouncer(zkConnectStr: String) extends FutureAnnouncer {
+class CuratorSDAnnouncer(zkConnectStr: String, registrationFormat: RegistrationFormat, backwardsCompatible: Boolean) extends FutureAnnouncer {
```

Comment:
One suggestion might be instead of using backwardsCompatible we can specify an array of enums for the registration info.  This way we can be explicit in terms of what we expect the behavior of the container to be and what formats we are supporting in production at any given time. 
 
Even more in the future we could probably convert registration enum into a class which we provide as a strategy pattern to convert tenant, company,